src/main/scala/org/wikimedia/analytics/hdfstools/HdfsRsyncCLI.scala

```diff
@@ -76,44 +111,95 @@ object HdfsRsyncCLI {
 
         opt[Unit]('v', "verbose")
             .optional()
-            .action((_, c) => c.copy(logLevel = Level.DEBUG))
-            .text("Add verbosity to logging (DEBUG)")
+            .action((_, c) => c.copy(applicationLogLevel = Level.DEBUG))
+            .text("Add verbosity to application logging (DEBUG)")
 
         opt[Unit]('q', "quiet")
             .optional()
-            .action((_, c) => c.copy(logLevel = Level.WARN))
-            .text("Remove verbosity from logging (WARN)")
+            .action((_, c) => c.copy(applicationLogLevel = Level.WARN))
+            .text("Remove verbosity from application logging (WARN)")
+
+        opt[Unit]("all-logs-info")
+            .optional()
+            .action((_, c) => c.copy(rootLogLevel = Level.INFO))
+            .text("Set root logger level to INFO (default: ERROR)")
+
+        opt[Unit]("all-logs-debug")
+            .optional()
+            .action((_, c) => c.copy(rootLogLevel = Level.DEBUG))
+            .text("Set root logger level to DEBUG (default: ERROR)")
+
+        opt[String]("log-file")
+            .optional()
+            .action((x, c) => c.copy(logFile = Some(x)))
+            .text("File to write logs instead of sending them to stdout")
+
+
 
         opt[Unit]('r', "recursive")
             .optional()
             .action((_, c) => c.copy(recurse = true))
             .text("Recurse into directories (default: false)")
 
-        opt[Unit]('p', "perms")
+        opt[Unit]('d', "dirs")
             .optional()
-            .action((_, c) => c.copy(preservePerms = true))
-            .text("Preserve permissions (default: false)")
+            .action((_, c) => c.copy(copyDirs = true))
+            .text("Copy directories without recursion (default: false)")
 
-        opt[Unit]('t', "times")
+        opt[Unit]('m', "prune-empty-dirs")
             .optional()
-            .action((_, c) => c.copy(preserveTimes = true))
-            .text("Preserve modification times (default: false)")
+            .action((_, c) => c.copy(pruneEmptyDirs = true))
+            .text("prune empty directory chains (default: false)")
 
-        opt[Long]("times-diff")
+
+
+        opt[Unit]("resolve-conflicts")
             .optional()
-            .action((x, c) => c.copy(acceptedTimesDiffMs = x))
-            .text("Milliseconds by which modificationTimes can differ and still be considered equal (default: 1000)")
+            .action((_, c) => c.copy(resolveConflicts = true))
+            .text("Resolve multi-source files/directories conflicts (default: false)\n" +
+                "\t\tNote: times and perms conflicts of merged-folders are resolved by default using the defined strategy.")
+
+        opt[Unit]("use-most-recent-time")
+            .optional()
+            .action((_, c) => c.copy(useMostRecentModifTimes = true))
+            .text("Use most recent modificationTime object instead of first listed src in conflicts (default: false)")
+
+
+
+        opt[Unit]("existing")
```

Comment:
Is this --existing-only, and is it incompatible with --ignore-existing?

Reply:
this is 'existing-only' indeed (I copied the option name from standard rsync). It is compatible with ignore-existing, as nothing prevents bot to be applied at the same time (no conflicting state), but having them both means nothing to do :)

Reply:
Ok let's keep as --existing then! TY!